app/models/payment.rb

```diff
@@ -0,0 +1,26 @@
+class Payment < ApplicationRecord
+  belongs_to :client
+  belongs_to :order
+
+  enum status: { pending: 0, paid: 5, fail: 7 }
+  validates :parcels, numericality: { greater_than_or_equal_to: 1 }, allow_blank: true
+
+  validates :parcels, :payment_method_id, presence: true
+  validates :invoice_token, presence: true, if: :paid?
+  validate :parcels_is_less_than_or_equal_to_max_parcels
+
+  def parcels_is_less_than_or_equal_to_max_parcels
+    payment_option = PaymentOption.find(payment_method_id)
+    return unless parcels.present? && payment_option.present? && parcels > payment_option.max_parcels
+
+    errors.add(:parcels, ' não pode ser maior que o máximo permitido pelo meio de pagamento')
+  end
+
+  def post_on_external_api
+    data = {invoice: {payment_method_id: self.payment_method_id, order_id: order.id, registration_number: client.cpf,
```

Comment:
Acho que no faraday você não passa o nome do nosso model, só os parametros direto
```suggestion
    data = {payment_method_id: self.payment_method_id, order_id: order.id, registration_number: client.cpf,
```

Reply:
ahh e a gente espera receber mais parametros também, mas não sei se é nesse PR
:status, :token, :package_id, :registration_number,
:insurance_company_id, :order_id, :payment_method_id,
:voucher, :parcels, :total_price,

Reply:
Obrigada, Petra! É aqui sim! Estávamos aguardando a inclusão desses parâmetros no código de vocês, então agora já podemos ajustar.

Reply:
Sobre o cometário da Petra sobre o model, na verdade isso nao tem a ver com o Faraday. Depende de como a API do outro lado espera os dados. É possível sim enviar os dados 'envelopados' com o nome do model ao redor. 

Ficaria como vcs fizeram, vcs postam { invoice : { atributo: valor, atributo: valor } } e do lado de lá é recebido no controller como: params[:invoice][:atributo].

Ter ou não o `invoice` (ou o nome de qualquer que seja o model) depende da documentação da API do outro lado.

Reply:
ops, o token é nosso mesmo!